#----------------- MPC param -----------------#
simulation:    false    # 是否在仿真模式下运行
perfect_simu:  false   # 是否为完美仿真，无模型误差和外部扰动
frequency:     100.0   # 控制频率（Hz）
ctrl_delay:    0.001    # 控制延迟时间（秒）
sfc_dis:       0.3     # 安全飞行走廊（SFC）的距离参数


thrust_limit:  0.99  # 推力限制，避免超出最大推力
hover_esti:    true    # 是否估计悬停推力
hover_perc:    0.7    # 估计的悬停推力百分比

yaw_ctrl_flag: false    # 是否启用偏航控制
yaw_gain:      2.0     # 偏航控制增益

goal_x:        0.0     # 目标位置 x 坐标
goal_y:        0.0     # 目标位置 y 坐标
goal_z:        1.0     # 目标位置 z 坐标

astar:
  resolution:  0.5     # A*算法的分辨率
  map_x_size:  12.0    # 地图的x尺寸
  map_y_size:  12.0    # 地图的y尺寸
  map_z_size:  12.0     # 地图的z尺寸
  expand_dyn:  0.5     # 动态障碍物扩张大小
  expand_fix:  0.4     # 固定障碍物扩张大小

fsm:
  ref_dis:     5      # 路径规划中参考点的距离
  path_dis:    0.1    # 路径规划中的路径点间隔

mpc:
  horizon:     15      # MPC的预测范围（时间步数）
  step:        0.1     # MPC的时间步长（秒）
  # ctrl_delay:  0     # 控制延迟，已在上面定义
  
 # MPC权重参数
  R_p:         1000.0  # 位置的权重
  R_v:         0.0     # 速度的权重
  R_a:         0.0     # 加速度的权重
  R_u:         0.0     # 控制输入的权重
  R_u_con:     0.2     # 控制输入连续性的权重
  R_pN:        2000.0  # 终端位置的权重
  R_vN:        200  # 终端速度的权重
  R_aN:        200  # 终端加速度的权重

  D_x:         0.0     # x轴阻尼
  D_y:         0.0     # y轴阻尼
  D_z:         0.0     # z轴阻尼

# 速度、加速度、控制输入的限制
  vx_min:     -10.0
  vy_min:     -10.0
  vz_min:     -10.0
  vx_max:      10.0
  vy_max:      10.0
  vz_max:      10.0

  ax_min:     -20.0
  ay_min:     -20.0
  az_min:     -10.0
  ax_max:      20.0
  ay_max:      20.0
  az_max:      20.0

  ux_min:     -100.0
  uy_min:     -100.0
  uz_min:     -100.0
  ux_max:      100.0
  uy_max:      100.0
  uz_max:      100.0

